File: closure-fizzbuzz/Clojure FizzBuzz/src/com/ziroby/kata/fizzbuzz/FizzBuzzer.clj
(ns com.ziroby.kata.fizzbuzz.FizzBuzzer
    (:gen-class
    :state state
    :init init
    :methods [
    [convert [int] String]
    ]))

(defn -init [] [[] (atom [])])

(defn -convert [this number]
  (cond
    (= 5 number) "buzz"
    (= 0 (mod number 3)) "fizz"
    :else (str number)))

File: fizzbuzz/src/com/ziroby/kata/fizzbuzz/FizzBuzzer.java
package com.ziroby.kata.fizzbuzz;

public class FizzBuzzer {

	public String buzz(int number) {
		if (isDivisibleBy(number, 3 * 5))
			return "fizzbuzz";
		if (isDivisibleBy(number, 3))
			return "fizz";
		if (isDivisibleBy(number, 5))
			return "buzz";
		
		return Integer.toString(number);
	}

	private boolean isDivisibleBy(int number, int divisor) {
		return number % divisor == 0;
	}

}
